### PR TITLE
Support `from_pretrained` for ASR and ENH

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR supports `from_pretrained` function for ASR and ENH.

```python
[ins] In [1]: from espnet2.bin.asr_inference import Speech2Text

[ins] In [2]: asr = Speech2Text.from_pretrained("Shinji Watanabe/laborotv_asr_train_asr_conformer2_latest33_raw_char_sp_valid.acc.ave")

[ins] In [3]: from espnet2.bin.enh_inference import SeparateSpeech

[ins] In [4]: ss = SeparateSpeech.from_pretrained("Chenda Li/wsj0_2mix_enh_train_enh_rnn_tf_raw_valid.si_snr.ave")
```

### CODE REVIEW

File: espnet2/bin/enh_inference.py [hunk 1]
```diff
@@ -253,6 +254,38 @@ def cal_permumation(self, ref_wavs, enh_wavs, criterion="si_snr"):
...
+
+            except ImportError:
+                logging.error(
+                    "`espnet_model_zoo` is not installed. "
+                    "Please install via `pip install -U espnet_model_zoo`."
+                )
+                raise
+            d = ModelDownloader()
+            train_config, model_file = d.download_and_unpack(enh_model_tag).values()
+            # FIXME: The arugment name is mismatched, it is better to rename.
```
Comment (kan-bayashi) [inline espnet2/bin/enh_inference.py:284]: @Emrys365 Here the arguments from downloader are different, which are defined in https://github.com/espnet/espnet/blob/fb75313bd1ddf027e075ea7197448c929b176f5d/espnet2/bin/pack.py#L24-L26
Maybe it is better to keep the same.

**Comment by kan-bayashi** [comment]
I also updated README in espnet_model_zoo.
https://github.com/espnet/espnet_model_zoo/pull/46

**Review by sw005320 — APPROVED** [review]
LGTM.
@YushiUeda, can you also work on the same thing for diarization?

**Comment by YushiUeda** [comment]
> @YushiUeda, can you also work on the same thing for diarization?
Sure! I will work on it right away.

**Review by ftshijt — COMMENTED** [review]
Looks cool from my side. Just two comments for discussion:

- I was using `init_param` for pre-training in the past, do you think we should somehow unify the interface with both?
- Could you please also add instances for some egs? people may not know we have provided this good function

**Comment by kan-bayashi** [comment]
> I was using init_param for pre-training in the past, do you think we should somehow unify the interface with both?

I think adding `from_pretrained` for each module is a little bit complicated and current `--init_param` option can deal with complex loading such as subset of parameters, excluding come parameters, and so on. Therefore, `from_pretrained` should be for only inference interface now.

> Could you please also add instances for some egs? people may not know we have provided this good function

For now I updated espnet_model_zoo instruction.
https://github.com/espnet/espnet_model_zoo/blob/5c20277fedab9377a421a587ab1fa2b10fd195df/README.md
I will update the usage in README in another PR.

**Review by ftshijt — APPROVED** [review]
LGTM! Many thanks

File: espnet2/bin/enh_inference.py (hunk 1 above)
Comment (Emrys365) [reply espnet2/bin/enh_inference.py:284]: Sorry for the mismatch. We should have kept consistency with other tasks. Could you remove the prefix "enh_" from all argument names in enh_inference.py?

Comment (kan-bayashi) [reply espnet2/bin/enh_inference.py:284]: I will do it in the next PR.

**Pull request merged**